orders.h

```diff
@@ -197,210 +202,220 @@ class SellOrder : public Order {
 };
 
 class BuildOrder : public Order {
   public:
-    BuildOrder();
-    ~BuildOrder();
+    BuildOrder(UnitId *target = nullptr) : Order(O_BUILD), target(target), needtocomplete(0) {}
+    BuildOrder(int needtocomplete) : Order(O_BUILD), target(nullptr), needtocomplete(needtocomplete) {}
+    virtual ~BuildOrder();
 
     UnitId *target;
-    int new_building;
+    int new_building = -1;
     int needtocomplete;
-    bool until_complete;
+    bool until_complete = false;
 };
 
 class SailOrder : public Order {
   public:
-    SailOrder();
-    ~SailOrder();
+    SailOrder() : Order(O_SAIL) {}
+    virtual ~SailOrder();
 
     std::list<MoveDir *> dirs;
 };
 
 class FindOrder : public Order {
   public:
-    FindOrder();
-    ~FindOrder();
+    FindOrder(int find) : Order(O_FIND), find(find) {}
+    virtual ~FindOrder() = default;
 
     int find;
 };
 
 class StealthOrder : public Order {
   public:
-    StealthOrder();
-    ~StealthOrder();
+    StealthOrder(int type = 0, UnitId *target = nullptr);
+    virtual ~StealthOrder();
 
     UnitId *target;
 };
 
 class StealOrder : public StealthOrder {
   public:
-    StealOrder();
-    ~StealOrder();
+    StealOrder(UnitId *target, int item)
+        : StealthOrder(O_STEAL, target), item(item) {}
+    virtual ~StealOrder() = default;
 
     int item;
 };
 
 class AssassinateOrder : public StealthOrder {
   public:
-    AssassinateOrder();
-    ~AssassinateOrder();
+      AssassinateOrder(UnitId * target = nullptr) : StealthOrder(O_ASSASSINATE, target) {}
+      virtual ~AssassinateOrder() = default;
 };
 
 class ForgetOrder : public Order {
   public:
-    ForgetOrder();
-    ~ForgetOrder();
+    ForgetOrder(int skill) : Order(O_FORGET), skill(skill) {}
+    virtual ~ForgetOrder() = default;
 
     int skill;
 };
 
 // Add class for exchange
 class ExchangeOrder : public Order {
   public:
-    ExchangeOrder();
-    ~ExchangeOrder();
+    ExchangeOrder(UnitId *target = nullptr) : Order(O_EXCHANGE), target(target) {};
+    virtual ~ExchangeOrder();
 
-    int giveItem;
-    int giveAmount;
-    int expectItem;
-    int expectAmount;
+    int giveItem = 0;
+    int giveAmount = 0;
+    int expectItem = 0;
+    int expectAmount = 0;
 
-    int exchangeStatus;
+    int exchangeStatus = -1;
 
     UnitId *target;
 };
 
 class TurnOrder : public Order {
   public:
-    TurnOrder();
-    ~TurnOrder();
+    TurnOrder(bool repeating = false) : Order(O_TURN), repeating(repeating) {};
+    virtual ~TurnOrder() = default;
+
     bool repeating;
     std::vector<std::string> turnOrders;
 };
 
 class CastOrder : public Order {
   public:
-    CastOrder();
-    ~CastOrder();
+    CastOrder(int spell, int level) : Order(O_CAST), spell(spell), level(level) {};
+    virtual ~CastOrder() = default;
 
     int spell;
     int level;
 };
 
 class CastMindOrder : public CastOrder {
   public:
-    CastMindOrder();
-    ~CastMindOrder();
+    CastMindOrder(int level, UnitId *id = nullptr);
```

Comment:
same with this and why keep it in the cpp file and not move it here?

Reply:
See above. it requires the spells enum from gamedata.h

Including gamedata.h in orders.h would have made any compilation unit that includes orders.h get recompiled anytime gamedata.h is touched. I wanted to avoid that.